launch/launch/exit_handler.py

```diff
@@ -100,3 +104,14 @@ def ignore_signal_exit_handler(context):
             context.task_state.returncode = 0
 
     default_exit_handler(context)
+
+
+def check_error_code_on_close(context):
+    """
+    Succeed if the process shuts down only if the error code is 0.
```

Comment:
This docblock is unclear to me. It should describe what it is actually doing. E.g. trigger teardown of launch if the task has a non-zero return code